libcaf_net/src/stream_socket.cpp

```diff
@@ -222,6 +222,10 @@ variant<size_t, sec> write(stream_socket x,
 
 variant<size_t, sec>
 check_stream_socket_io_res(std::make_signed<size_t>::type res) {
```

Comment:
`send` and `receive` return values mean different things. I think it would be a good idea to add a flag `rw` to this check function to distinguish between a `send` or a `receive` result and handle them separately.